File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/common/uicomponent/audioeffect/service/AudioEffectService.java
package com.trtc.uikit.livekit.common.uicomponent.audioeffect.service;

import com.tencent.liteav.audio.TXAudioEffectManager;
import com.tencent.trtc.TRTCCloud;
import com.trtc.uikit.livekit.common.uicomponent.audioeffect.store.AudioEffectSateFactory;
import com.trtc.uikit.livekit.common.uicomponent.audioeffect.store.AudioEffectState;
import com.trtc.uikit.livekit.common.utils.LiveKitLog;

public class AudioEffectService {

    public static final String TAG = "AudioEffectService";

    private final String           mRoomId;
    private final TRTCCloud        mTRTCCloud;
    public        AudioEffectState mAudioEffectState;

    public AudioEffectService(String roomId, TRTCCloud trtcCloud) {
        mRoomId = roomId;
        mTRTCCloud = trtcCloud;
        mAudioEffectState = AudioEffectSateFactory.getState(roomId);
    }

    public void setVoiceChangerType(TXAudioEffectManager.TXVoiceChangerType type) {
        LiveKitLog.info(TAG + "[" + mRoomId + "] setVoiceChangerType:[type:" + type
                + "]");
        mTRTCCloud.getAudioEffectManager().setVoiceChangerType(type);
        mAudioEffectState.changerType.set(type);
    }

    public void setVoiceReverbType(TXAudioEffectManager.TXVoiceReverbType type) {
        LiveKitLog.info(TAG + "[" + mRoomId + "] setVoiceReverbType:[type:" + type + "]");
        mTRTCCloud.getAudioEffectManager().setVoiceReverbType(type);
        mAudioEffectState.reverbType.set(type);
    }

    public void enableVoiceEarMonitor(boolean enable) {
        LiveKitLog.info(TAG + "[" + mRoomId + "] enableVoiceEarMonitor:[enable:" + enable + "]");
        mTRTCCloud.getAudioEffectManager().enableVoiceEarMonitor(enable);
        mAudioEffectState.enableVoiceEarMonitor.set(enable);
    }

    public void setMusicVolume(int volume) {
        LiveKitLog.info(TAG + "[" + mRoomId + "] setMusicVolume:[volume:" + volume + "]");
        mTRTCCloud.getAudioEffectManager().setAllMusicVolume(volume);
        mAudioEffectState.musicVolume.set(volume);
    }

    public void setVoiceEarMonitorVolume(int volume) {
        LiveKitLog.info(TAG + "[" + mRoomId + "] setVoiceEarMonitorVolume:[volume:" + volume + "]");
        mTRTCCloud.getAudioEffectManager().setVoiceEarMonitorVolume(volume);
        mAudioEffectState.earMonitorVolume.set(volume);
    }

    public void setVoiceVolume(int volume) {
        LiveKitLog.info(TAG + "[" + mRoomId + "] setVoiceVolume:[volume:" + volume + "]");
        mTRTCCloud.getAudioEffectManager().setVoiceCaptureVolume(volume);
        mAudioEffectState.voiceVolume.set(volume);
    }
}

File: Flutter/livekit/lib/common/ui_component/audio_effect/service/audio_effect_service.dart
import 'package:tencent_live_uikit/common/ui_component/audio_effect/store/audio_effect_state.dart';
import 'package:tencent_live_uikit/manager/live_controller.dart';

class AudioEffectService {
  final LiveController liveController;
  final AudioEffectState audioEffectState;

  const AudioEffectService({required this.liveController, required this.audioEffectState});

  void setVoiceChangerType(int type) async {
    await liveController.liveService.setVoiceChangerType(type);
    audioEffectState.changerType.value = type;
  }

  void setVoiceReverbType(int type) {
    liveController.liveService.setVoiceReverbType(type);
    audioEffectState.reverbType.value = type;
  }

  void enableVoiceEarMonitor(bool enable) {
    liveController.liveService.enableVoiceEarMonitor(enable);
    audioEffectState.enableVoiceEarMonitor.value = enable;
  }

  void setMusicVolume(int volume) {
    liveController.liveService.setAllMusicVolume(volume);
    audioEffectState.musicVolume.value = volume;
  }

  void setVoiceEarMonitorVolume(int volume) {
    liveController.liveService.setVoiceEarMonitorVolume(volume);
    audioEffectState.earMonitorVolume.value = volume;
  }

  void setVoiceVolume(int volume) {
    liveController.liveService.setVoiceCaptureVolume(volume);
    audioEffectState.voiceVolume.value = volume;
  }
}
